### PR TITLE
Refresh device labels after gUM grant (makes them show in HTTP in Firefox)

### PR DESCRIPTION
**Pull request by jan-ivar**
Makes camera and mic labels appear
* instantly on first use without requiring user to refresh the page in Chrome and Firefox.
* in Firefox in normal flow (wo/requiring user to be on HTTPS *and* explicitly choosing "Always Allow").

### CODE REVIEW

**Comment by samdutton** [comment]
Minor JSHint nit: 

src/content/devices/input-output/js/main.js

     47 |      return n.value == values[i];
                                ^ Expected '===' and instead saw '=='.

File: src/content/devices/input-output/js/main.js [hunk 1]
```diff
@@ -12,8 +12,17 @@ var videoElement = document.querySelector('video');
 var audioInputSelect = document.querySelector('select#audioSource');
 var audioOutputSelect = document.querySelector('select#audioOutput');
 var videoSelect = document.querySelector('select#videoSource');
+var selectors = [audioInputSelect, audioOutputSelect, videoSelect];
 
 function gotDevices(deviceInfos) {
+  // Handles being called several times to update labels. Preserve values.
+  var values = selectors.map(function(select) { return select.value; });
```
Comment (samdutton) [inline src/content/devices/input-output/js/main.js]: Google style nit: 

    var values = selectors.map(function(select) { 
        return select.value; 
    });

File: src/content/devices/input-output/js/main.js [hunk 2]
```diff
@@ -33,17 +42,19 @@ function gotDevices(deviceInfos) {
       console.log('Some other kind of source/device: ', deviceInfo);
     }
   }
+  selectors.forEach(function(select, i) {
+    if (Array.prototype.slice.call(select.childNodes).some(function(n) {
+      return n.value == values[i];
```
Comment (samdutton) [inline src/content/devices/input-output/js/main.js]: ===

Comment (jan-ivar) [inline src/content/devices/input-output/js/main.js]: ?

**Comment by jan-ivar** [comment]
I hadn't done npm install in a while. Thanks for the tip!

**Comment by samdutton** [comment]
If you think this slows down commits too much, we can change to pre-push:

    githooks: {
      all: {
        'pre-commit': 'csslint htmlhint jscs jshint'
      }
    }

**Comment by samdutton** [comment]
BTW: On first opening in Chrome and Firefox I get mic and camera labels with the existing https://webrtc.github.io/samples/src/content/devices/input-output.

**Comment by jan-ivar** [comment]
@samdutton Do you see "microphone 1" and "camera 1", or actual hardware-identifying device labels?

Are you prompted for permission?

**Comment by samdutton** [comment]
I see labels.

**Comment by jan-ivar** [comment]
@samdutton WFM. Are you prompted for permission? If no:

In Chrome, make sure you don't have persistent permissions granted, by clicking "Manage Media Settings..." under the camera icon in the URL bar, remove `https://webrtc.github.io:443`, and hit Done.

In Firefox, make sure you don't have persistent permissions granted, find `webrtc.github.io` in about:permissions and hit "Forget about this site".

If yes:

In Firefox, close other tabs open that may be sharing microphone with webrtc.github.io (unfotunately, there's no hardware light for microphones). There's also a sharing indicator top-center on the desktop if this is happening, that can be used to track down the offending tab.

In Chrome, I would be surprised if this happened.

If you're still seeing labels, I'll need some STRs and other info.

**Comment by KaptenJansson** [comment]
Nice PR, this works fine for me.

Also I can confirm when I delete my permissions and using the existing site, I do NOT get the real device labels unless I refresh the page after approving the first time on HTTPS.

@samdutton could you confirm?

**Comment by KaptenJansson** [comment]
Rerunning the test to see if this works with the input-output test. LGTM if it works.

**Pull request merged**